### PR TITLE
Implement multi-process message visualisation.

### PR DESCRIPTION
**Pull request by MichaelMauderer**
### Pull Request Description

Implements a visualization that is integrated with our GUI profiling visualization for the multiprocess data implemented in #3395 

https://user-images.githubusercontent.com/1428930/165915395-c850c7b2-1cc5-4eb0-8f21-37565d113b1e.mp4

The visualization shows a horizontal line for Engine, Language Server and GUI and renders arrows for each message passed between them. Information about the message is revealed on hover.

### Important Notes
* this PR refactors the tooltip mechanism. Note that this has not been in active use anywhere else, as tooltips for node received a custom implementation and the tooltip that was previously implemented was used nowhere else yet. 

[ci no changelog needed]

### Checklist

Please include the following checklist in your PR:

- [x] The documentation has been updated if necessary.
- [x] All code conforms to the [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md), [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md), and [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guides.
- All code has been tested:
  - [x] Unit tests have been written where possible.
  - [x] If GUI codebase was changed: Enso GUI was tested when built using BOTH `./run dist` and `./run watch`.

### CODE REVIEW

File: lib/rust/ensogl/component/sequence-diagram/src/labeled_line.rs [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+// ===========
+
+#[derive(Clone, Copy, Debug)]
+pub enum Cap {
+    Start,
+    End,
+}
+
+impl Default for Cap {
+    fn default() -> Self {
```
Comment (wdanilo) [inline lib/rust/ensogl/component/sequence-diagram/src/labeled_line.rs]: A lot of missing docs here - please be sure that linter is enabled. Is Clippy telling about it? If not, why?

**Review by wdanilo — CHANGES_REQUESTED** [review]

File: lib/rust/ensogl/core/src/gui/style.rs [hunk 2]
```diff
@@ -66,6 +66,10 @@ impl<T> StyleValue<T> {
 #[macro_export]
 /// Create struct called `Style` that can hold updatable style information.
 macro_rules! define_style {( $( $(#$meta:tt)* $field:ident : $field_type:ty),* $(,)? ) => {
+
+    #[allow(unused_imports)]
+    use $crate::prelude::*;
```
Comment (kazcw) [inline lib/rust/ensogl/core/src/gui/style.rs]: `use` inside a macro imports into the callee's namespace and may cause surprises.

File: lib/rust/ensogl/component/sequence-diagram/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+        let actor_lines = default();
+        let message_lines = default();
+        let origin_x = default();
+        Self { app, display_object, actor_lines, message_lines, origin_x }
+    }
+}
+
+impl Model {
+    fn set_profile(&self, profiles: &[Profile<Metadata>]) {
+        if let [profile_a, profile_b] = profiles {
```
Comment (kazcw) [inline lib/rust/ensogl/component/sequence-diagram/src/lib.rs:103]: Should have some handling for the `else` case (at least clear any previous diagram).

File: lib/rust/ensogl/component/sequence-diagram/src/lib.rs [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+    }
+}
+
+
+
+// =============
+// === Model ===
+// =============
+
+/// /// Internal model of the SequenceDiagram.
```
Comment (kazcw) [inline lib/rust/ensogl/component/sequence-diagram/src/lib.rs]: ///

**Review by kazcw — APPROVED** [review]

**Review by wdanilo — APPROVED** [review]

**Pull request merged**